client/src/components/RocketProfilePopup.tsx

```diff
@@ -54,32 +70,56 @@ const RocketProfilePopup: React.FC<RocketProfileProps> = (props: RocketProfilePr
 	const [editMode, setEditMode] = useState<boolean>(props.rocketProfileId !== '-1');
 
 	useEffect(() => {
-		setEditMode(props.rocketProfileId !== '')
+		setEditMode(props.rocketProfileId !== '');
 		// Reset State
 		setHeight(NaN);
 		setMass(NaN);
 		setRocketClass('');
 		setMotor('');
 		setName('');
 		setMotorType('');
+		setComponentList([]);
+		componentIdToName.current = {};
 		async function getRocketDetails() {
-			if (props.rocketProfileId) {
-				const response = await axios.get<IRocketDetails>(
+			if (props.rocketProfileId || newRocketData.current) {
+				const rocketResponse = await axios.get<IRocketDetails>(
 					`http://127.0.0.1:9090/rocket/${props.rocketProfileId}`
 				);
-				const data = response.data.result;
+				const componentResponse = await axios.get<IComponentDetails>(`http://127.0.0.1:9090/component`);
+				const data = rocketResponse.data.result;
+				const componentData = componentResponse.data.results;
+				componentData.forEach((component: IComponentModel) => {
+					componentIdToName.current[component._id] = component.Name;
+				});
+				setComponentList(componentData);
```

Comment:
This code is repeated in refresh. Could probably move it into a fucntion called `getComponents`. You could also use the useCallback hook to minize the number of calls being made. It'd look something likde this:

```tsx
const getComponents = useCallback(async () => {
    // api call
}, []);

```